tests/ops/test_qubit_ops.py

```diff
@@ -3496,3 +3499,269 @@ def f(state):
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+                0.0 + 0.0j,
+            ]
+        )
+
+        @qml.qnode(dev)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            qml.PauliX(wires=1)
+            qml.OrbitalRotation(phi, wires=[0, 1, 2, 3])
+
+            return qml.state()
+
+        assert np.allclose(state, circuit(np.pi / 2))
+
+    @pytest.mark.parametrize("diff_method", ["parameter-shift", "backprop"])
+    @pytest.mark.parametrize(
+        ("phi"),
+        [
+            (-0.1),
+            (0.1),
+        ],
+    )
+    def test_autograd_grad(self, phi, diff_method):
+        """Tests that gradients are computed correctly using the
+        autograd interface"""
+
+        pytest.importorskip("autograd")
+
+        dev = qml.device("default.qubit.autograd", wires=4)
+
+        @qml.qnode(dev, interface="autograd", diff_method=diff_method)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            qml.PauliX(wires=1)
+            qml.OrbitalRotation(phi, wires=[0, 1, 2, 3])
+
+            return qml.expval(qml.PauliZ(0))
+
+        assert np.allclose(qml.grad(circuit)(phi), np.sin(phi))
+
+    @pytest.mark.parametrize("diff_method", ["parameter-shift", "backprop"])
+    @pytest.mark.parametrize(
+        ("phi"),
+        [(-0.1), (0.1)],
+    )
+    def test_tf_grad(self, phi, diff_method):
+        """Tests that gradients are computed correctly using the
+        tensorflow interface"""
+
+        tf = pytest.importorskip("tensorflow")
+        dev = qml.device("default.qubit.tf", wires=4)
+
+        @qml.qnode(dev, interface="tf", diff_method=diff_method)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            qml.PauliX(wires=1)
+            qml.OrbitalRotation(phi, wires=[0, 1, 2, 3])
+            return qml.expval(qml.PauliZ(0))
+
+        phi_t = tf.Variable(phi, dtype=tf.float64)
+        with tf.GradientTape() as tape:
+            res = circuit(phi_t)
+
+        grad = tape.gradient(res, phi_t)
+
+        assert np.allclose(grad, np.sin(phi))
+
+    @pytest.mark.parametrize("diff_method", ["parameter-shift", "backprop"])
+    @pytest.mark.parametrize(
+        ("phi"),
+        [
+            (-0.1),
+            (0.1),
+        ],
+    )
+    def test_jax_grad(self, phi, diff_method):
+        """Tests that gradients and operations are computed correctly using the
+        jax interface"""
+
+        if diff_method == "parameter-shift":
+            pytest.skip("JAX support for the parameter-shift method is still TBD")
+
+        jax = pytest.importorskip("jax")
+
+        dev = qml.device("default.qubit.jax", wires=4)
+
+        @qml.qnode(dev, interface="jax", diff_method=diff_method)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            qml.PauliX(wires=1)
+            qml.OrbitalRotation(phi, wires=[0, 1, 2, 3])
+            return qml.expval(qml.PauliZ(0))
+
+        assert np.allclose(jax.grad(circuit)(phi), np.sin(phi))
```

Comment:
@obliviateandsurrender could you add a PyTorch test here as well? Aside from that, great tests!

Reply:
Hi @obliviateandsurrender, nice catch -- pytorch can be very finicky with how matrices are defined.

No need to modify, but I'm curious if using `qml.math.stack` to first stack each row of the array, and then stack all rows, would have worked with PyTorch?

Reply:
Hey @josh146, you're right. Using `qml.math.stack` indeed works! I think it would be much better to use this instead of the full decomposition.

Reply:
Awesome! let me know when the master merge conflict is resolved, and I will give it another review :)

Reply:
Hello, the merge conflict with the master is resolved now.